megatron/initialize.py

```diff
@@ -205,7 +206,7 @@ def _initialize_distributed():
             else:
                 args.local_rank = device
 
-        torch.cuda.set_device(device) 
+            torch.cuda.set_device(device) 
```

Comment:
Double-check the indentation.

Reply:
This indentation is a bug fix to only do so when device_count > 0. Added a comment to clarify.

Reply:
Looks good.